Implement frac and lerp builtins

tgjones opened the pull request:
I don't know which naming convention you prefer - I've gone with the HLSL names for these functions, because HLSL is what I know best. Let me know if you'd prefer to use the GLSL or MetalSL names for either or both of these functions.

feliwir commented:
I think you have some comment names wrong in ShaderBuiltins.cs

mellinoe commented:
`Lerp` is actually already supported if you just call [`Vector3.Lerp`](https://msdn.microsoft.com/en-us/library/system.numerics.vector3.lerp(v=vs.111).aspx). I'm not sure it's worth adding because of that. The `Frac` looks good to me.

mellinoe commented on src/ShaderGen.Primitives/ShaderBuiltins.cs:
```diff
@@ -43,6 +43,18 @@ public static Vector4 Sample(TextureCubeResource texture, SamplerResource sample
         public static Vector3 Cos(Vector3 value) => throw new ShaderBuiltinException();
         public static Vector4 Cos(Vector4 value) => throw new ShaderBuiltinException();
 
+        // Cos
```
As @feliwir  mentioned, this comment is wrong .

tgjones commented:
Oops, I've fixed the comments.

I thought `Lerp` was useful to have for scalar floats, and then for consistency I added the vector overloads too (same as `Vector3.Abs`, for example, which has both the `Vector3` methods as well as `ShaderBuiltins` methods).

mellinoe commented:
@tgjones Actually, I think it's fine, since there is no Math.Lerp method. Some of the stuff was added to `ShaderBuiltins` before I started adding mappings for "regular" types (like `Math`, `MathF`, `Vector2/3/4`, etc.), so some of them are duplicated like Abs. I think it's fine in this case though, since there would be no way to expose the scalar version as you've mentioned.

mellinoe commented:
Looks good, thanks for the contribution!
